src/ray/object_manager/object_manager.cc

```diff
@@ -387,10 +387,18 @@ void ObjectManager::Push(const ObjectID &object_id, const NodeID &node_id) {
 
     UniqueID push_id = UniqueID::FromRandom();
     push_manager_->StartPush(node_id, object_id, num_chunks, [=](int64_t chunk_id) {
-      SendObjectChunk(push_id, object_id, owner_address, node_id, data_size,
-                      metadata_size, chunk_id, rpc_client, [=](const Status &status) {
-                        push_manager_->OnChunkComplete(node_id, object_id);
-                      });
+      rpc_service_.post([=]() {
```

Comment:
[=] will unnecessary copy object_info, which is probably not a big deal. But why don't we just capture the exact values we need here?
```
rpc_service_.post([push_id, object_id, owner_address, node_id, data_size,metadata_size, chunk_id, rpc_client]()
```

Reply:
I believe it only copies the variables that are used: https://en.cppreference.com/w/cpp/language/lambda

Reply:
Oh that's right. I didn't know about that. It LGTM then!